caravel/assets/javascripts/SqlLab/components/VisualizeModal.jsx

```diff
@@ -26,6 +26,19 @@ class VisualizeModal extends React.Component {
       columns: {},
       hints: [],
     };
+    // update columns if possible
+    this.componentWillMount();
```

Comment:
I wouldn't call react's lifecycle methods directly as the framework typically handles them. Just refactor what's in `componentWillMount` into it's own method and call that one from both the constructor and `componentWillMount `